3_visualize/src/diagnostic_log_fcn.R

```diff
@@ -0,0 +1,47 @@
+#' # Define diagnostic_log() function
+#' 
+#' The `diagnostic_log()` function creates a text file of model diagnostic results.
+#' 
+#' @param input_file chr, file path of the subfolder where processed data is saved
+#' 
+#' @param output_file chr, file path where the final diagnostic log will be saved
+#' 
+#' @param model_type chr vector, list of model types by IDs 
+#' 
+#' @param exper_num numerical vector, list of experimental temperatures (numerical)
+#'
+diagnostic_log <- function(input_file, output_file, model_type, exper_num){
+  
+  # Load in the processed data
+  input_data <- read.csv(input_file)
+  
+  # Save the model diagnostics
+  render_data <- NULL
+  
+  for(mm in 1:length(model_type)){ #for each model type
+    for(ee in 1:length(exper_num)){ # and each exper id 
+      
+      # filter data based on model type and exper id
+      filterdata <- input_data[input_data$model_type == model_type[mm] & 
+                                 input_data$n_prof == exper_num[ee],]
+      
+      # calculate mean RMSE and store in list
+      templist <- list(filterdata %>% pull(rmse) %>% mean %>% round(2))
+      names(templist) <- paste0(model_type[mm], "_", exper_num[ee], "mean")
+      render_data <- append(x = render_data, values = templist)
+      }
+  }
```

Comment:
Nice crack at this one. Two comments:
* I would take it one step further and probably pull this thing out and make it a helper function
* appending to an empty table can be a big time sink for large datasets. This pattern works well here, but would probably need a bit more thought if you were to scale it up.